src/test/commons/com/trzewik/jdbc/db/AccountCreator.groovy

```diff
@@ -0,0 +1,26 @@
+package com.trzewik.jdbc.db
+
+trait AccountCreator {
+    Account createAccount(AccountBuilder builder = new AccountBuilder()){
+        new Account(
+            builder.userId,
+            builder.username,
+            builder.email
+        )
+    }
+
+    Account createAccount(long userId, Account account = createAccount()){
+        new Account(
+            userId,
+            account.username,
+            account.email
+        )
+    }
+
+    static class AccountBuilder{
```

Comment:
jaki ten obiekt ma związek z design-patternem o nazwie `Builder` który jest w suffixie jego nazwy?

Reply:
zmieniam nazwę na `AccountCreator`. I tak, można to tę klasę zastąpić mapą przekazywaną do metody `createAccount(Map account)`, ale:
- nie będę mógł zmienić tylko jedne defaultowej wartości - albo biorę wszystko z defaultu, albo nic
- intelliJ nie będzie mi podpowiadał co tam mogę wstawić
- zmiana nazwy pola jest mniej bolesna w przypadku klasy niż mapy (łatwiej będzie to refaktorować)
- można mieć kilka konstruktorów, które będą tworzyć dany obiekt w różny sposób :D i będziemy mieli to wszystko w jednym miejscu